pllug-presentation-system/presentation-data/domdocumentdivider.h

```diff
@@ -0,0 +1,32 @@
+#ifndef DOMDOCUMENTDIVIDER_H
+#define DOMDOCUMENTDIVIDER_H
+
+#include <QStringList>
+#include <QString>
+#include <QFile>
+#include <QVector>
+#include <QtWidgets/QMessageBox>
+#include <QDebug>
+#include <QDomDocument>
+
+#include "documentimportstrategy.h"
+#include "slide.h"
+#include "presentationelementfactory.h"
+#include "presentation.h"
```

Comment:
Use forward declaration in header files, if this is a pointer or reference type, and move all includes to *.cpp files.
You may read article about forward declaration [here](http://www.umich.edu/~eecs381/handouts/IncompleteDeclarations.pdf).